google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/Firestore.java

```diff
@@ -93,7 +94,18 @@
    * @param documentReferences List of Document References to fetch.
    */
   @Nonnull
-  ApiFuture<List<DocumentSnapshot>> getAll(final DocumentReference... documentReferences);
+  ApiFuture<List<DocumentSnapshot>> getAll(@Nonnull DocumentReference... documentReferences);
+
+  /**
+   * Retrieves multiple documents from Firestore, while optionally applying a field mask to reduce
+   * the amount of data transmitted.
+   *
+   * @param documentReferences Array with Document References to fetch.
+   * @param fieldMask If set, specifies the subset of fields to return.
+   */
+  @Nonnull
+  ApiFuture<List<DocumentSnapshot>> getAll(
+      @Nonnull DocumentReference[] documentReferences, @Nullable FieldMask fieldMask);
```

Comment:
just curious: Do we prefer array over list?

Reply:
I think we might prefer a `List` here, but the Array matches the getAll(...) varargs method. If a user passes an array, they can omit the last argument and fall back to the varargs method.

I'm super torn on this and prefer Lists in general.

Reply:
Ah I see the tradeoffs. I'm OK with this now.